services/provider/server/cephrbdmirror.go

```diff
@@ -0,0 +1,43 @@
+package server
+
+import (
+	"context"
+
+	rookCephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const rBDMirrorName = "rbd-mirror"
+
+type cephRBDMirrorManager struct {
+	client    client.Client
+	namespace string
+}
+
+func newCephRBDMirrorManager(cl client.Client, namespace string) (*cephRBDMirrorManager, error) {
+	return &cephRBDMirrorManager{
+		client:    cl,
+		namespace: namespace,
+	}, nil
+}
+
+func (c *cephRBDMirrorManager) Create(ctx context.Context) error {
+
+	cephRBDMirror := &rookCephv1.CephRBDMirror{}
+	cephRBDMirror.Name = rBDMirrorName
+	cephRBDMirror.Namespace = c.namespace
+	err := c.client.Get(ctx, client.ObjectKeyFromObject(cephRBDMirror), cephRBDMirror)
```

Comment:
any reason in making two calls rather than a single create?

Reply:
We need one rbdMirror for the cluster, hence I am checking if it already exists, and create it only if it doesn't